Dockerfile

```diff
@@ -1,6 +1,22 @@
+
+FROM golang:1.15.5-alpine AS builder
+
+RUN apk add --update --no-cache ca-certificates bash make gcc musl-dev git openssh wget curl
+
+WORKDIR /go/src/bifrost
+
+COPY go.mod .
+COPY go.sum .
+RUN go mod download
+
+COPY . .
+RUN CGO_ENABLED=0 make build
+
+#####
+
 FROM alpine
 
-ADD bifrost /bifrost
-ADD config/config.json config/config.json
+COPY --from=builder /go/src/bifrost/bifrost /bifrost
+COPY --from=builder /go/src/bifrost/config/config.json config/config.json
 
```

Comment:
A question that I have is there is no `EXPOSE` directive here. How is the port exposed?

Reply:
Is it any different from defining when running the container? (i.e. `-p`)

Reply:
That's only to bind the internal port to an external port right? But the expose directive has to be there first. Maybe I am wrong though.

Reply:
It [seems](https://docs.docker.com/engine/reference/builder/#expose) it has some documentation purposes as well. It doesn't need to be there to being able to bind a port.

Reply:
Oh wow, TIL. Thanks @isacikgoz :)